### PR TITLE
Add explicit invariant when ReactDOM is loaded without React

### LINKED ISSUE #10406: Show a better message if React is not loaded
Currently if React is not loaded, ReactDOM shows this:

screen shot 2017-08-08 at 3 36 52 pm

We should throw a nicer error message in the entry point I think.

### PR DESCRIPTION
**Pull request by aweary**
Resolves #10406

### CODE REVIEW

File: src/renderers/dom/shared/checkReact.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @providesModule checkReact
+ * @flow
+ */
+
+var invariant = require('fbjs/lib/invariant');
+
+invariant(
+  require('react'),
+  'ReactDOM was loaded without React. Make sure you load ' +
+    'the React package before loading ReactDOM.',
```
Comment (aweary) [inline src/renderers/dom/shared/checkReact.js]: I'm not a big fan of this copy, I'm open to suggestions 😄

File: src/renderers/dom/shared/checkReact.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule checkReact
+ * @flow
+ */
+
+var invariant = require('fbjs/lib/invariant');
+
+invariant(
+  require('react'),
```
Comment (gaearon) [inline src/renderers/dom/shared/checkReact.js]: Let's extract this to a variable to be consistent.

File: src/renderers/dom/shared/checkReact.js (hunk 1 above)
Comment (gaearon) [inline src/renderers/dom/shared/checkReact.js]: Maybe "ReactDOM was loaded before React."

Comment (gaearon) [inline src/renderers/dom/shared/checkReact.js]: The other sentence looks good to me.

**Pull request merged**